descartes_utilities/include/descartes_utilities/ros_conversions.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2016, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ *  ros_conversions.h
+ *
+ *  Created on: Feb 28, 2016
+ *  Author: Jonathan Meyer
+ */
+
+#ifndef DESCARTES_ROS_CONVERSIONS_H
+#define DESCARTES_ROS_CONVERSIONS_H
+
+#include <trajectory_msgs/JointTrajectory.h>
+#include <descartes_core/trajectory_pt.h>
+
+namespace descartes_utilities
+{
+  /**
+   * @brief Converts a sequence of Descartes joint trajectory points to ROS trajectory points.
```

Comment:
@shaun-edwards We can fit a constrained cubic spline to each joint, and come up with smooth (if possibly out of range) velocities/accels for the entire trajectory without modifying the timing. I played with the concept using Eigen and got something going. I know everyone will want a library, so I've been looking:

I looked at the [KDL](http://docs.ros.org/hydro/api/orocos_kdl/html/classKDL_1_1VelocityProfile__Spline.html) spline function you mentioned, but it appears to apply to just one 'segment' between two points.

There's a very full featured solution in [ecl](http://wiki.ros.org/ecl_geometry/Tutorials/Splines) that will probably work, but it seems like quite the dependency.

The spline_smoother from the old arm nav package also had similar routines but I don't know if there's a catkin version.

@gavanderhoorn Are you aware of a good fit?